File: backend/app/routers/public_config.py
"""Unauthenticated by design: the frontend needs to know which backend mode is active before a
user has logged in at all, so it knows whether to show an Appwrite-flavoured or a local login and
registration form. Nothing here is sensitive; BACKEND_MODE is not a secret.
"""

from fastapi import APIRouter

from .. import config

router = APIRouter(prefix="/api/public-config", tags=["public-config"])


@router.get("")
def get_public_config():
    return {"backend_mode": config.BACKEND_MODE}

File: frontend/src/api/publicConfig.ts
import type { PublicConfig } from "../types";

/** Unauthenticated: tells the app which backend is active before anyone has logged in. */
export async function getPublicConfig(): Promise<PublicConfig> {
  const res = await fetch("/api/public-config");
  if (!res.ok) throw new Error(`failed to load server configuration: ${res.status}`);
  return res.json();
}
